### PR TITLE
fix(app registration process): fix wrong api calling to download documents

### PR DESCRIPTION
**Pull request by kunalgaurav-bmw**
## Description

fix wrong api is called in the registration process of an app

Changelog entry:

```
- **App Registration Process**:
  - fixed wrong api calling to download documents [#1386](https://github.com/eclipse-tractusx/portal-frontend/pull/1386)
```

## Why

Correct api should be called for download documents in the registration process of an app to make it functional.

## Issue

https://github.com/eclipse-tractusx/portal-frontend/issues/1226

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/admin/Dev%20Process/How%20to%20contribute.md#commits-branches-and-pull-requests-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx [hunk 1]
```diff
@@ -450,6 +451,23 @@ export default function AppMarketCard() {
...
 
+  const handleDownload = async (documentName: string, documentId: string) => {
+    if (fetchDocumentById)
+      try {
+        const response = await fetchDocumentById({
+          appId,
+          documentId,
+        }).unwrap()
+
+        const fileType = response.headers.get('content-type')
```
Comment (manojava-gk) [inline src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx:462]: these two lines are duplicated in many place. Can you move this to utils?

**Review by manojava-gk — COMMENTED** [review]
Avoid code duplication

File: src/components/shared/basic/ReleaseProcess/AppPage/index.tsx [hunk 2]
```diff
@@ -466,6 +470,45 @@ export default function AppPage() {
...
+        const fileType = response.headers.get('content-type')
+        const file = response.data
+
+        download(file, fileType, documentName)
+      } catch (error) {
+        console.error(error, 'ERROR WHILE FETCHING DOCUMENT')
+      }
+  }
+
+  const handleDownloadFile = async (
```
[the remark this answers is not in the document]
Comment (kunalgaurav-bmw) [reply src/components/shared/basic/ReleaseProcess/AppPage/index.tsx]: Done

File: src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx (hunk 1 above)
Comment (kunalgaurav-bmw) [reply src/components/shared/basic/ReleaseProcess/AppMarketCard/index.tsx]: Done

File: src/features/appManagement/types.ts [hunk 3]
```diff
@@ -83,6 +83,11 @@ export interface DescriptionState {
   shortDescription: string
 }
 
+export interface FileState {
```
Comment (manojava-gk) [inline src/features/appManagement/types.ts:86]: Can you move this to ```FileUtils``` file?

File: src/components/shared/basic/ReleaseProcess/AppPage/index.tsx [hunk 4]
```diff
@@ -467,6 +472,37 @@ export default function AppPage() {
...
   }
 
+  const handleDownload = async (
+    documentName: FileState[] | string,
+    documentId?: string
+  ) => {
+    let docId: string
+    let docName: string
+
+    if (Array.isArray(documentName)) {
```
Comment (manojava-gk) [inline src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:482]: can you refactor it to a single statement?

ex - ```const documentName = Array.isArray(documentName) ? documentName[0].name : documentName```

File: src/features/appManagement/types.ts (hunk 3 above)
Comment (kunalgaurav-bmw) [reply src/features/appManagement/types.ts]: Moved

File: src/components/shared/basic/ReleaseProcess/AppPage/index.tsx (hunk 4 above)
Comment (kunalgaurav-bmw) [reply src/components/shared/basic/ReleaseProcess/AppPage/index.tsx]: Changes done

File: src/components/shared/basic/ReleaseProcess/AppPage/index.tsx [hunk 5]
```diff
@@ -68,6 +68,9 @@ import {
 import { ButtonLabelTypes } from '..'
 import { PrivacyPolicyType } from 'features/adminBoard/adminBoardApiSlice'
 import { phone } from 'phone'
+import { useFetchDocumentByIdMutation } from 'features/apps/apiSlice'
+import { download } from 'utils/downloadUtils'
+import { type FileState, extractFileData } from 'utils/fileUtils'
```
Comment (manojava-gk) [inline src/components/shared/basic/ReleaseProcess/AppPage/index.tsx:73]: please align with @lavanya-bmw on case usage

File: src/utils/fileUtils.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  id: string
+  name: string
+}
+
+interface FileResponse {
+  headers: Headers
+  data: Blob
+}
+
+export const extractFileData = (response: FileResponse) => {
```
Comment (manojava-gk) [inline src/utils/fileUtils.ts:30]: please align with @lavanya-bmw on case usage

**Review by lavanya-bmw — APPROVED** [review]
lgtm

**Review by manojava-gk — APPROVED** [review]

**Review by oyo — APPROVED** [review]

**Pull request merged**